common/src/chain/tokens/tokens_utils.rs

```diff
@@ -13,37 +13,20 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use super::{OutputValue, TokenData, TokenId};
+use super::TokenId;
 use crate::{
-    chain::{Transaction, TxOutput},
+    chain::{Transaction, TxInput, TxOutput},
     primitives::id::hash_encoded,
 };
 
 pub fn token_id(tx: &Transaction) -> Option<TokenId> {
     Some(hash_encoded(tx.inputs().get(0)?))
 }
 
-pub fn is_tokens_issuance(output_value: &OutputValue) -> bool {
-    match output_value {
-        OutputValue::Coin(_) => false,
-        OutputValue::Token(token_data) => match **token_data {
-            TokenData::TokenIssuance(_) | TokenData::NftIssuance(_) => true,
-            TokenData::TokenTransfer(_) => false,
-        },
-    }
+pub fn token_id_from_input0(input0: &TxInput) -> TokenId {
```

Comment:
I think this should be a private function. Exposing it makes it possible to abuse it. The function `token_id` must be used instead.

Reply:
fixed